libraries/amqp/include/eosio/amqp/amqp_handler.hpp

```diff
@@ -94,7 +94,8 @@ class amqp {
    /// Do not call from lambda's passed to publish or constructor e.g. on_error
    void stop() {
       std::unique_lock<std::mutex> g(mtx_);
-      if( handler_ ) {
+      if( !stopped_ ) {
+         stopped_ = true;
```

Comment:
Now, `stop()` does not guarantee that the thread pool is stopped before it returns.  I seem to recall that being important.

Reply:
Gave up on making this thread safe and reverted back to comments describing non-thread-safe. It is used appropriately everywhere currently.
Removed mutex and used boost post to make all calls to on_error be from the amqp thread.